File: app/src/main/java/com/lucasurbas/counter/app/di/AppModule.java
package com.lucasurbas.counter.app.di;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    private final Context appContext;

    public AppModule(Context context) {
        this.appContext = context.getApplicationContext();
    }

    @Singleton
    @Provides
    Context getApplicationContext() {
        return appContext;
    }

}

File: app/src/main/java/com/lucasurbas/counter/app/di/AppModule.kt
package com.lucasurbas.counter.app.di

import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule(context: Context) {

    private val appContext: Context

    init {
        appContext = context.applicationContext
    }

    @Singleton
    @Provides
    internal fun provideApplicationContext(): Context {
        return appContext
    }
}
